File: shift_negative.c
#include <stdio.h>
#include <stdlib.h>

void main() 
{
    unsigned char uc = 0x00;
    
    uc = 0x01 >> 1;    
    printf("uc >> 1 = %02x\n", uc); 

    uc = 0x01;
    printf("uc = %02x\n", uc); 

    uc = 0x01 >> -1;
    printf("uc >> -1 = %02x\n", uc); 
} 

File: shift_negative.rs
fn main() {
    let mut uc:u8 = 0x00;
    
    uc = 0x01 >> 1;
    println!("uc >> 1= {}", uc);

    uc = 0x01;
    println!("uc = {}", uc);

    uc = 0x01 >> -1;
    println!("uc >> -1 = {}", uc);
    
}